schema/cassandra/cadence/versioned/v0.29/processing_queue_states.cql

```diff
@@ -0,0 +1,14 @@
+CREATE TYPE domain_filter (
+    domain_ids    set<uuid>,
+    reverse_match boolean,
+    );
+
+CREATE TYPE processing_queue_state (
+    processing_queue_level  int,
+    ack_level               bigint,
+    max_level               bigint,
+    domain_filter           frozen<domain_filter>,
+);
```

Comment:
We will need another type for timer queue as the type of ack_level and max_level field should be timestamp for timer queue.

Reply:
Chatted offline about this. I'll keep the field as bigint here to avoid using the timestamp type, but will create a new data interface struct for timer queue state and handle the conversion between time and bigint in the persistence.

Reply:
Another idea it that we can store a blob for all processing queue states in a shard, check here: https://github.com/uber/cadence/pull/3304/files?file-filters%5B%5D=.cql&file-filters%5B%5D=.go#diff-6f7954af286de899bc2234fbbbd083fbR23